internal/storage/bucket_handle.go

```diff
@@ -141,6 +141,13 @@ func (bh *bucketHandle) StatObject(ctx context.Context,
 		err = fmt.Errorf("error in fetching object attributes: %w", err)
 		return
 	}
+	// TODO(anushkadhn): read only for unfinalized object. To be done after PR#3365 is merged
+	if req.ReadWhileStat {
```

Comment:
so many changes to get readWhileStat to this file.

Can we simply check if the bucket is zonal and enableRapidAppends are true, make a call. You need to pass enableRapidAppends flag to bucket_handle only during initialization.